splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java

```diff
@@ -191,15 +190,19 @@ SplunkRum initialize(
                     });
         }
 
-        otelRumBuilder.addInstrumentation(
-                instrumentedApplication -> {
-                    SlowRenderingDetector slowRenderingDetector =
-                            buildSlowRenderingDetector(
-                                    instrumentedApplication
-                                            .getOpenTelemetrySdk()
-                                            .getTracer(SplunkRum.RUM_TRACER_NAME));
-                    slowRenderingDetector.start(instrumentedApplication.getApplication());
-                });
+        if (builder.slowRenderingDetectionEnabled) {
+            otelRumBuilder.addInstrumentation(
+                    instrumentedApplication -> {
+                        SlowRenderingDetector.builder()
+                                .setSlowRenderingDetectionPollInterval(
+                                        builder.slowRenderingDetectionPollInterval)
+                                .build()
+                                .installOn(instrumentedApplication);
+                        initializationEvents.add(
+                                new RumInitializer.InitializationEvent(
+                                        "slowRenderingDetectorInitialized", timingClock.now()));
+                    });
+        }
```

Comment:
If this was factored out to a named method like `initializeSlowRenderingDetection(otelRumBuilder)` or something, you could return early from the if and save some indent depth. It's getting a little burly in there.

Reply:
Done. I also extracted similar private methods for other instrumentations.